extensions/jobrunrpro.md

```diff
@@ -0,0 +1,37 @@
+# JobRunrPro
+
+The purpose of this extension is to leverage some features only available in the Pro version of [JobRunr](https://www.jobrunr.io/en/documentation/pro/).
+Only the Pro version allows to search existing jobs by status and label. This is required for the deadline manager to implement the `cancelAll` methods.
+The [deadline managers](../axon-framework/deadlines/deadline-managers.md) section has more in depth information on deadline managers.
+Although jobs created with the non-pro `DeadlineManager` will be eligible to be canceled, this is only true when they were created with the `4.8` or later versions.
+Jobs created with the `4.7` version are missing the correct labels and will not be found when trying to cancel them.
+
+## Spring usage
+
+For spring usage, be sure to include the starters, both of JobRunr Pro and the extension. The deadline manager should be available by autowiring.
+It can by used in aggregates and sagas using something like:
```

Comment:
For Aggregate's this would be a suboptimal suggestion, as an Aggregate is a prototype bean. Instead it's better to simply wire the `DeadlineManager` in the message handler. So, something like this:

```java
@CommandHandler
public void handle(SomeCommand command, @Autowired DeadlineManager deadlineManager) { ... }
```

For sagas, setter-based injection would work. But, similarly, you can also use parameter-based injection (done through Axon's `ParameterResolvers`).

So, to accompany both solutions, perhaps we can use a code block.
Doing so, we allow users to select the use case (Saga or Aggregate), and show the appropriate samples.

Reply:
updated